File: app/ts/src/auth/modules/none.ts
import type { AuthContext, AuthResult } from "../../core/types.ts";
import type { AuthModule, AuthDeps } from "../types.ts";

export class NoneAuth implements AuthModule {
  static readonly typeName = "none";

  static fromConfig(
    _config: Record<string, unknown>,
    _deps: AuthDeps,
  ): NoneAuth {
    return new NoneAuth();
  }

  async apply(_ctx: AuthContext): Promise<AuthResult> {
    return {};
  }

  async invalidate(): Promise<void> {}

  isRejection(_statusCode: number, _body: unknown): boolean {
    return false;
  }
}

File: app/python/auth/modules/none_.py
from __future__ import annotations

from typing import Any

import httpx
from redis.asyncio import Redis

from app.python.auth.base import AuthContext, AuthResult


class NoneAuth:
    """Auth module that injects nothing; for open/public upstreams."""

    name = "none"

    def __init__(self) -> None:
        pass

    @classmethod
    def from_config(
        cls, config: dict[str, Any], *, redis: Redis, http: httpx.AsyncClient
    ) -> NoneAuth:
        return cls()

    async def apply(self, ctx: AuthContext) -> AuthResult:
        return AuthResult()

    async def invalidate(self) -> None:
        pass

    def is_rejection(self, status_code: int, body: dict | None) -> bool:
        return False
